src/components/Coupon/CouponItem.tsx

```diff
@@ -0,0 +1,118 @@
+import { css } from '@emotion/react';
+import { ChangeEvent } from 'react';
+import { useRecoilState, useRecoilValue } from 'recoil';
+
+import Checkbox from '../common/Checkbox';
+
+import { MINIMUM_FREE_SHIPPING_AMOUNT } from '@/constants/cart';
+import { MAX_SELECTED_COUPON_LENGTH } from '@/constants/coupon';
+import { Coupon } from '@/types/coupon';
+import { dateFormat, timeFormat } from '@/utils/format';
+import { orderTotalPriceState } from '@recoil/cartItems/selectors';
+import { selectedCouponsState } from '@recoil/coupon/atom';
+import { applicableCouponSelector, isMaxSelectedCouponsSelector } from '@recoil/coupon/selector';
+
+interface CouponItemProps {
+  coupon: Coupon;
+}
+
+export default function CouponItem({ coupon }: CouponItemProps) {
+  const [selectedCoupons, setSelectedCoupons] = useRecoilState(selectedCouponsState);
+  const isCouponApplicable = useRecoilValue(applicableCouponSelector(coupon.code));
+  const isMaxSelectedCoupons = useRecoilValue(isMaxSelectedCouponsSelector);
+  const orderTotalPrice = useRecoilValue(orderTotalPriceState);
+
+  const isShippingFreeCouponDisabled =
+    coupon.discountType === 'freeShipping' && orderTotalPrice >= MINIMUM_FREE_SHIPPING_AMOUNT;
+
+  const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.checked && isMaxSelectedCoupons) {
+      alert(`쿠폰은 최대 ${MAX_SELECTED_COUPON_LENGTH}개만 선택 가능합니다!`);
+      return;
+    }
+
+    if (e.target.checked) {
+      setSelectedCoupons([...selectedCoupons, coupon.code]);
+    } else {
+      if (selectedCoupons.includes(coupon.code))
```

Comment:
단순 궁금) `else if() {}`가 아니라 `else { if() }` 로 의도해주신 구문일까용 ?

Reply:
아하, 이렇게 쓰는 것은 처음봐서 궁금해서 질문드렸어요. 복잡한 코드가 아니라 코드 파악에 어려움은 없었찌만, 뒤에 다른 코드를 추가해주실 계획이 있으시려나 하는 생각도 살짝 들었습니다. 🤔

Reply:
아하 확실히 그렇게 느껴질 수도 있을 것 같네요!

생각해보니, 쿠폰은 체크 해제할 때, 선택되지 않았을 리가 없으니, 이렇게 if문을 제거해주어도 괜찮겠네요..!


```suggestion
    } else {
      setSelectedCoupons(selectedCoupons.filter((code) => coupon.code !== code));
```